jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt

```diff
@@ -17,24 +18,41 @@ class DeployServerlessApplicationAction : AnAction(
         AwsIcons.Resources.LAMBDA_FUNCTION) {
     private var templateYamlRegex = Regex("template\\.y[a]?ml", RegexOption.IGNORE_CASE)
 
-    override fun actionPerformed(e: AnActionEvent?) {
+    override fun actionPerformed(e: AnActionEvent) {
 
-        val project = e?.getRequiredData(PlatformDataKeys.PROJECT) ?: throw Exception("Unable to determine project")
+        val project = e.getRequiredData(PlatformDataKeys.PROJECT)
 
-        val virtualFiles = e.getData(PlatformDataKeys.VIRTUAL_FILE_ARRAY) ?: throw Exception("Could not detect template file")
-        val samTemplateFile = virtualFiles[0]
+        val samTemplateFile = getSamTemplateFile(e) ?: throw Exception("Could not detect template file")
         val template = CloudFormationTemplate.parse(project, samTemplateFile)
 
         // TODO : Validate the template file (this is likely too slow to do in update())
 
         DeployServerlessApplicationDialog(project, template.parameters()).show()
     }
 
-    override fun update(e: AnActionEvent?) {
+    override fun update(e: AnActionEvent) {
         super.update(e)
 
-        val virtualFiles = e?.getData(PlatformDataKeys.VIRTUAL_FILE_ARRAY)
+        e.presentation.isVisible = getSamTemplateFile(e) != null
+    }
+
+    /**
+     * Determines the relevant Sam Template, returns null if one can't be found.
+     */
+    private fun getSamTemplateFile(e: AnActionEvent): VirtualFile? {
+        val virtualFiles = e.getData(PlatformDataKeys.VIRTUAL_FILE_ARRAY) ?: return null
+        val virtualFile = virtualFiles.singleOrNull() ?: return null
+
+        if (templateYamlRegex.matches(virtualFile.name)) {
+            return virtualFile
+        }
+
+        // If the module node was selected, see if there is a template file in the top level folder
+        val project = e.getData(PlatformDataKeys.PROJECT) ?: return null
+        if (virtualFile == project.baseDir) {
```

Comment:
This method is deprecated: https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/openapi/project/Project.java#L44

Reply:
Also you should be looking at the Module, not project